### PR TITLE
fix(nvd-cve-osv): add a 2% deletion threshold check

### PR DESCRIPTION
**Pull request by andrewpollock**
After the events of #2678, add in a threshold check of the number of GCS objects that would get deleted and abort if more than 10% would be.

Follow
https://google.github.io/styleguide/shellguide.html#which-shell-to-use for setting shell options.

### CODE REVIEW

File: vulnfeeds/cmd/nvd-cve-osv/run_cve_to_osv_generation.sh [hunk 1]
```diff
@@ -56,6 +60,14 @@ for (( YEAR = $(date +%Y) ; YEAR >= ${FIRST_INSCOPE_YEAR} ; YEAR-- )); do
 done
 
 # Copy results to GCS bucket.
+objs_present=$(gsutil ls "${OSV_OUTPUT_GCS_PATH}" | wc -l)
+objs_deleted=$(gsutil -m rsync -n -c -d "${WORK_DIR}/nvd2osv/gcs_stage" "${OSV_OUTPUT_GCS_PATH}" 2>&1 | grep -c "Would delete")
+
+if (( objs_deleted < objs_present * 0.1 )); then
```
Comment (oliverchang) [inline vulnfeeds/cmd/nvd-cve-osv/run_cve_to_osv_generation.sh:66]: should this be > instead?

Comment (andrewpollock) [reply vulnfeeds/cmd/nvd-cve-osv/run_cve_to_osv_generation.sh]: Yes. Fixed this, and many other things surfaced from rigorous manual testing in [bb87c97](https://github.com/google/osv.dev/pull/2732/commits/bb87c979c8c58eba16e56c223f263076c542266c)

**Comment by another-rex** [comment]
Given #2730 would it make sense to also make this 2%? 10% of all CVE records seem like a lot of entries.

**Comment by andrewpollock** [comment]
> Given #2730 would it make sense to also make this 2%? 10% of all CVE records seem like a lot of entries.

Fair call. I've defaulted it to this and made it able to be overridden in case this value turns out to be troublesome.

**Review by oliverchang — APPROVED** [review]

**Pull request merged**